modules/ismp/testsuite/src/lib.rs

```diff
@@ -402,7 +434,188 @@ pub fn check_request_source_and_destination() {}
+        nonce: 0,
+        from: vec![0u8; 32],
+        to: vec![0u8; 32],
+        timeout_timestamp: 0,
+        data: vec![0u8; 64],
+        gas_limit: 0,
+    };
+
+    let dispatch_request = DispatchRequest::Post(dispatch_post);
+    dispatcher
+        .dispatch_request(dispatch_request, [0; 32].into(), 0u32.into())
+        .unwrap();
+
+    // Request message handling check for source and destination chain
+    let request_message = Message::Request(RequestMessage {
+        requests: vec![post.clone()],
+        proof: Proof { height: intermediate_state.height, proof: vec![] },
+        signer: vec![0u8; 32],
+    });
+
+    handle_incoming_message(host, request_message).unwrap();
+
+    let request = Request::Post(post);
+    // Fetch commitment from storage
+    let commitment = hash_request::<H>(&request);
+    host.request_commitment(commitment)
+        .map_err(|_| "Expected Request commitment to be found in storage")?;
+    let post = Post {
+        source: host.host_state_machine(),
+        dest: intermediate_state.height.id.state_id,
+        nonce: 0,
+        from: vec![0u8; 32],
+        to: vec![0u8; 32],
+        timeout_timestamp: 0,
+        data: vec![0u8; 64],
+        gas_limit: 0,
+    };
+    let response =
+        PostResponse { post: post.clone(), response: vec![], timeout_timestamp: 0, gas_limit: 0 };
+    dispatcher
+        .dispatch_response(response.clone(), [0; 32].into(), 0u32.into())
+        .unwrap();
+
+    // Fetch commitment from storage
+    let commitment = hash_post_response::<H>(&response.clone());
+    host.response_commitment(commitment)
+        .map_err(|_| "Expected Request commitment to be found in storage")?;
+    // Response message handling check
+    let msg = ResponseMessage {
+        datagram: RequestResponse::Response(vec![Response::Post(response.clone())]),
+        proof: Proof { height: invalid_state_commitment.height, proof: vec![] },
+        signer: vec![],
+    };
+    assert_ne!(response.source_chain(), msg.proof.height.id.state_id);
+    let res = handle_incoming_message(host, Message::Response(msg.clone()));
+    assert!(matches!(res, Err(..)));
+    Ok(())
+}
 
 /// Check that proxies can dispatch requests & responses.
-pub fn sanity_check_for_proxies() {}
+pub fn sanity_check_for_proxies<H, D>(host: &H, dispatcher: &D) -> Result<(), &'static str>
+where
+    H: IsmpHost,
+    D: IsmpDispatcher,
+    D::Account: From<[u8; 32]>,
+    D::Balance: From<u32>,
+{
+    let proxy_state_commitment = setup_mock_proxy_client(host);
+    // Set the previous update time
+    let challenge_period = host.challenge_period(mock_proxy_consensus_state_id()).unwrap();
+    let previous_update_time = host.timestamp() - (challenge_period * 2);
+    host.store_consensus_update_time(mock_proxy_consensus_state_id(), previous_update_time)
+        .unwrap();
+    host.store_state_machine_update_time(proxy_state_commitment.height, previous_update_time)
+        .unwrap();
+
```

Comment:
Assert that a proxy is configured and the request destination != host state machine id.